controllers/keystoneapi_controller.go

```diff
@@ -526,9 +519,12 @@ func (r *KeystoneAPIReconciler) reconcileNormal(ctx context.Context, instance *k
 	// create hash over all the different input resources to identify if any those changed
 	// and a restart/recreate is required.
 	//
-	inputHash, err := r.createHashOfInputHashes(ctx, instance, configMapVars)
+	inputHash, hashChanged, err := r.createHashOfInputHashes(ctx, instance, configMapVars)
 	if err != nil {
 		return ctrl.Result{}, err
+	} else if hashChanged {
+		// Hash changed and instance status should be updated, so we need to return and reconcile again
+		return ctrl.Result{}, nil
```

Comment:
With this we'll only reconcile on the next event, right? for running jobs we can expect one to come. just wondering if there can be a situation in one of the bellow `return ctrl.Result{}, nil` that this might not happen? It might make sense to set `Requeue` to true ? [1]

[1] https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.1/pkg/reconcile#Result

Reply:
Yeah, good point.  We should force the requeue whenever we update the status and need it to save and then continue reconciling again.

Reply:
don't we get reconciled because we changed our own Status?

Reply:
Is relying on hashChanged evaluation safe here? I see it takes the instance as an argument, so wouldn't the mentioned earlier read-after-write (stale reads) race condition happening here?

Reply:
@gibizer Yes, my Github comment is outdated and the code comment is what is proper here

Reply:
thank you for explaining